project/UI/CargoProgress.gd

```diff
@@ -1,42 +1,42 @@
 extends ProgressBar
 
-export var Ore: PackedScene = preload("res://World/Ores/IronOre.tscn")
+@export var Ore: PackedScene = preload("res://World/Ores/IronOre.tscn")
 
-onready var arc_bottom := $ArcBottom
-onready var arc_top := $ArcTop
-onready var fill := $Fill
-onready var tween := $Tween
-onready var anim_player := $AnimationPlayer
-onready var audio_unload: AudioStreamPlayer = $AudioUnload
+@onready var arc_bottom := $ArcBottom
+@onready var arc_top := $ArcTop
+@onready var fill := $Fill
+@onready var tween : Tween
+@onready var anim_player := $AnimationPlayer
+@onready var audio_unload: AudioStreamPlayer = $AudioUnload
 
 var docked_position := Vector2.ZERO
 
 var _player_is_mining := false
 
 
 func _ready() -> void:
-	Events.connect("docked", self, "_on_Events_docked")
-	Events.connect("mine_started", self, "_on_Events_mine_started")
-	Events.connect("mine_finished", self, "_on_Events_mine_finished")
-	self.connect("value_changed", self, "_on_value_changed")
+	Events.connect("docked", Callable(self, "_on_Events_docked"))
+	Events.connect("mine_started", Callable(self, "_on_Events_mine_started"))
+	Events.connect("mine_finished", Callable(self, "_on_Events_mine_finished"))
+	self.connect("value_changed", Callable(self, "_on_value_changed"))
```

Comment:
In Godot 4, the idiomatic syntax would be e.g. `Events.docked.connect(_on_Events_docked)`. It gives you static checks, unlike when using strings.

Then, for single-use functions, I'd refactor to turn them into lambda functions. You can just cut and paste the function into the connect() call as an argument so it's an easy refactor. Here's an example:

```gdscript
Events.docked.connect(func _on_Events_docked(docking_point: DockingPoint) -> void:
	docked_position = docking_point.get_global_transform_with_canvas().origin)
```

These two changes combined give you both static checks and keep the code grouped in one place instead of needing you to jump back and forth to read it.

Reply:
atm I'm focusing on getting the project working as it used to. There's many things that are broken due to substantial changes from Godot 3 to 4, notably Tweens, Viewports, and class inheritance, that the auto-update tools can only do so much about (in these cases, nothing). The things that are working after the auto update, without much work on my part, I'm committing as-is.

I'll work on cleaning things up and updating to the newer features of GDScript once everything's working